### PR TITLE
Migration Mobbing: Validate contacts inside of the case entity

### PR DESCRIPTION
**Pull request by ericsorenson**
This is a precursor for strict validation and migration validation errors.

### CODE REVIEW

File: shared/src/utilities/JoiValidationDecorator.js [hunk 1]
```diff
@@ -72,6 +72,15 @@ function getFormattedValidationErrors(entity) {
     errors = getFormattedValidationErrorsHelper(entity);
   }
   if (errors) {
+    for (const key of Object.keys(errors)) {
+      if (
+        // remove unhelpful error messages from contact validations
+        typeof errors[key] == 'string' &&
+        errors[key].endsWith('does not match any of the allowed types')
+      ) {
+        errors[key] = undefined;
```
Comment (kkoskelin) [inline shared/src/utilities/JoiValidationDecorator.js]: _nit_: You might consider `delete errors[key];`

Comment (kkoskelin) [reply shared/src/utilities/JoiValidationDecorator.js]: We should also get some tests around this.

**Review by kkoskelin — APPROVED** [review]
Once test coverage is complete, 👍

File: shared/src/utilities/JoiValidationDecorator.js (hunk 1 above)
Comment (ericsorenson) [reply shared/src/utilities/JoiValidationDecorator.js]: Good call.

**Review by kkoskelin — APPROVED** [review]
prs-actually

**Pull request merged**